arduino/src/arduino.cpp

```diff
@@ -0,0 +1,285 @@
+/**
+ * @file arduino.cpp
+ * @author yarin
+ * @brief  the arduino<-> pi protocol arduino object
+ * @version 0.1
+ * @date 2020-11-21
+ * @copyright Copyright (c) 2020
+ * 
+ */
+
+#include "arduino.h"
+#define RATE 9600
+#define TIMEOUT 10000
+#define READLENGTH 100
+#define SUCCESS 1
+#define DATALEN 1000
+#define COMMENDHEADER "3"
+#define FAILURE -1
+#define NULLCHAR '\0'
+#define CHECKBUFFSIZE 64
+
+/*!
+ * @brief Destroy the Arduino:: Arduino object 
+ * probably meaningless 
+ */
+Arduino::~Arduino()
+{
+    this->serial.closeDevice();
+}
+/**
+ * @brief Construct a new Arduino:: Arduino object
+ * 
+ */
+Arduino::Arduino(std::string portName, unsigned int buadRate, unsigned int timeout) : portName(portName), buadRate(buadRate), timeout(timeout)
+{
+}
+/**
+ * @brief getter, will return an array of files that the class uses
+ * to output of the sensors
+ * 
+ * @return std::vector<std::string> an array of string 
+ */
+std::vector<std::string> Arduino::getFiles()
+{
+    return this->files;
+}
+/**
+ * @brief getter for the timeout of the serial port
+ * 
+ * @return int the time to abandon the serial communication
+ */
+int Arduino::getTimout()
+{
+    return this->timeout;
+}
+/**
+ * @brief setter of the timeout
+ * 
+ * @param timeout int the time to abandon the serial communication
+ */
+void Arduino::setTimeout(int timeout)
+{
+    this->timeout = timeout;
+}
+/**
+ * @brief check the connection to the arduino and report.
+ */
+Arduino::ERROR_ARDUINO Arduino::checkConnection()
+{
+    if (!this->isOpen)
+        return Arduino::ERROR_ARDUINO::SUCCSESS;
+    char helper = this->serial.openDevice(portName.c_str(), buadRate);
+    if (helper != SUCCESS)
+    {
+        serial.closeDevice();
+        return Arduino::ERROR_ARDUINO::SUCCSESS;
+    }
+    serial.closeDevice();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief check the connection for debugging
+ */
+void Arduino::checkConnectionToConsole()
+{
+    if (checkConnection() != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        std::cout << "there was a problem with the connection" << std::endl;
+    }
+    std::cout << "the connection is successful!-" << std::endl;
+}
+/**
+ * @brief add file to the files array
+ * @param file add a file to the array of files
+ */
+
+void Arduino::addFile(std::string file)
+{
+    this->files.push_back(file);
+}
+/*
+ * @brief will calculate the sum of a string
+ * I forgot that I found this function on the web so I made 
+ * a simple version of this using std::string
+ * this one will probably be used in the arduino
```

Comment:
you dont need to explain way the functions is here but what the function do.
also you dont need to tell me how it dose what it dose you just need to tell me what it should do

Reply:
fixed